redisson/src/main/java/org/redisson/api/MapOptions.java

```diff
@@ -89,7 +96,11 @@ public static <K, V> MapOptions<K, V> defaults() {
      * @return MapOptions instance
      */
     public MapOptions<K, V> writer(MapWriter<K, V> writer) {
-        this.writer = writer;
+        if (writerRetryAttempts ==0) {
```

Comment:
I thought you will remove that check since you were right - writerRetryAttempts can be set later

Reply:
I didn't remove it because I didn't get your confirmation. Now I got it and changed it back.